File: be/src/main/java/com/artsykov/backend/service/CompanyService.java
package com.artsykov.backend.service;

import com.artsykov.backend.entity.CompanyEntity;
import com.artsykov.backend.model.CompanyPageModel;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public interface CompanyService {
    CompanyEntity saveCompany(CompanyEntity companyEntity);

    CompanyEntity findCompanyByLogInInfId(int logInInfId);

    CompanyEntity findCompany(int companyId);

    CompanyEntity saveWallet(CompanyEntity companyEntity);

    CompanyPageModel findAllByPage(int pageNumber, int amount);

    void changeStatus(CompanyEntity companyEntity);
}

File: frontend/src/app/services/company.service.ts
import {Observable} from "rxjs";
import {companyModel} from "../modules/models/companyModel";
import {companyOrErrorsModel} from "../modules/models/companyOrErrorsModel";
import {CompanyPageModel} from "../modules/models/CompanyPageModel";

export interface CompanyService {
  findAllCompanies(): Observable<companyModel[]>;

  saveCompany(company: companyModel): Observable<companyOrErrorsModel>;

  saveCompanyWallet(company: companyModel): Observable<companyOrErrorsModel>;

  findCompanyByLogInInfId(logInInfId: number): Observable<companyModel>;

  findCompanyById(id: number): Observable<companyModel>;

  findAllByPage(pageNumber: number, amount: number): Observable<CompanyPageModel>;

  changeStatus(companyModel: companyModel): Observable<companyOrErrorsModel>;
}
